packages/neon_framework/packages/notifications_push_repository/lib/src/notifications_push_repository.dart

```diff
@@ -0,0 +1,243 @@
+import 'dart:async';
+import 'dart:typed_data';
+
+import 'package:account_repository/account_repository.dart';
+import 'package:built_collection/built_collection.dart';
+import 'package:collection/collection.dart';
+import 'package:crypton/crypton.dart';
+import 'package:logging/logging.dart';
+import 'package:nextcloud/nextcloud.dart';
+import 'package:nextcloud/notifications.dart' as notifications;
+import 'package:notifications_push_repository/src/models/models.dart';
+import 'package:notifications_push_repository/src/notifications_push_storage.dart';
+import 'package:notifications_push_repository/src/utils/encryption.dart';
+import 'package:unifiedpush/unifiedpush.dart';
+
+/// Signature of the callback triggered by UnifiedPush when a new push notification is received.
+typedef OnMessageCallback = void Function(Uint8List message, String accountID);
+
+final _log = Logger('NotificationsPushRepository');
+
+/// A repository for managing push notifications and subscriptions.
+final class NotificationsPushRepository {
+  /// Creates a new [NotificationsPushRepository].
+  NotificationsPushRepository({
+    required AccountRepository accountRepository,
+    required NotificationsPushStorage storage,
+    required OnMessageCallback onMessage,
+  })  : _accountRepository = accountRepository,
+        _storage = storage,
+        _onMessage = onMessage;
+
+  final AccountRepository _accountRepository;
+  final NotificationsPushStorage _storage;
+  final OnMessageCallback _onMessage;
+  var _initialized = false;
+  StreamSubscription<BuiltList<Account>>? _accountsListener;
+  late final RSAPrivateKey _privateKey;
+  late BuiltList<Account> _accounts;
```

Comment:
Please remove this interlocked state and just fetch it from the account_repo on demand.